chain-signatures/node/src/backlog.rs

```diff
@@ -283,23 +704,79 @@ mod tests {
     async fn test_backlog_chain_isolation() {
         let backlog = Backlog::new();
 
-        let tx_eth = create_test_tx(1, PendingRequestStatus::PendingExecution);
-        let tx_sol = create_test_tx(2, PendingRequestStatus::PendingExecution);
-        let tx_near = create_test_tx(3, PendingRequestStatus::PendingExecution);
+        let tx_eth = create_test_tx(1, PendingRequestStatus::AwaitingResponse);
+        let tx_sol = create_test_tx(2, PendingRequestStatus::AwaitingResponse);
+        let tx_near = create_test_tx(3, PendingRequestStatus::AwaitingResponse);
 
         let sign_id_eth = SignId::new(tx_eth.request_id);
         let sign_id_sol = SignId::new(tx_sol.request_id);
         let sign_id_near = SignId::new(tx_near.request_id);
 
+        let program_id = Pubkey::new_unique();
+
         // Insert into different chains
         backlog
-            .insert(Chain::Ethereum, sign_id_eth, tx_eth.clone())
+            .insert(
+                Chain::Ethereum,
+                sign_id_eth,
+                BacklogTransaction::Bidirectional(tx_eth.clone()),
+                SignRequestType::SignBidirectional(SignBidirectionalEvent {
+                    sender: Default::default(),
+                    serialized_transaction: vec![],
+                    dest: "ethereum".to_string(),
+                    caip2_id: "eip155:1".to_string(),
+                    key_version: 0,
+                    deposit: 0,
+                    path: "".to_string(),
+                    algo: "".to_string(),
+                    params: "".to_string(),
+                    program_id,
+                    output_deserialization_schema: vec![],
+                    respond_serialization_schema: vec![],
+                }),
```

Comment:
nitpick on style: It would be slightly easier to read (and less code duplication) to use a common constructor and a spread operator, something like:

```suggestion
                SignRequestType::SignBidirectional(SignBidirectionalEvent {
                    dest: "ethereum".to_string(),
                    caip2_id: "eip155:1".to_string(),
                    program_id,
                    ..SignBidirectionalEvent::empty()
                }),
```

Instead of `::empty()`, you could potentially even use `#[derive(Default)]` on `SignBidirectionalEvent` but personally I would give the constructor a different name, since the result isn't a valid default event.